subgraphs/network/schema.graphql

```diff
@@ -0,0 +1,326 @@
+  blocks: [Block!]! @derivedFrom(field: "day")
+}
+
+type HourlySnapshot @entity {
+  " { number of hours since UNIX epoch time } "
+  id: ID!
+
+  " ID of the network this snapshot belongs to "
+  network: Network!
+
+  " Number of cumulative unique authors (see type Author for explanation) "
+  cumulativeUniqueAuthors: Int!
+
+  " Number of unique hourly active authors "
+  hourlyActiveAuthors: Int!
+
+  " Block height at this hour (ie, blockNumber) "
+  blockHeight: Int!
+
+  " Timestamp of this snapshot "
+  timestamp: BigInt!
+
+  " Blocks created during this hour "
+  hourlyBlocks: Int!
+
+  " Blocks/day calculated at the last block this hour "
+  blocksPerDay: BigDecimal!
+
+  " Total difficulty up until this hour "
+  cumulativeDifficulty: BigInt!
+
+  " Difficulty during this hour "
+  hourlyDifficulty: BigInt!
+
+  " Average difficulty for this hour "
+  hourlyMeanDifficulty: BigDecimal!
+
+  " Sum of all gas used during this hour "
+  hourlyCumulativeGasUsed: BigInt!
+
+  " Sum of all gas limit during this hour "
+  hourlyCumulativeGasLimit: BigInt!
+
+  " Mean of gas used per block during this hour "
+  hourlyMeanGasUsed: BigDecimal!
+
+  " Mean of gas limit per block during this hour "
+  hourlyMeanGasLimit: BigDecimal!
+
+  " Total amount of fees burned "
+  cumulativeBurntFees: BigInt!
+
+  " Amount of burnt fees for this hour "
+  hourlyBurntFees: BigInt!
+
+  " Total rewards emitted up until this hour "
+  cumulativeRewards: BigInt
+
+  " Rewards emitted for this hour (ie, new supply this hour) "
+  hourlyRewards: BigInt
+
+  " First timestamp of this entity (used to calculate averages) "
+  firstTimestamp: BigInt!
+
+  " Hourly mean block interval time in seconds during this hour "
+  hourlyMeanBlockInterval: BigDecimal!
+
+  " Cumulative size of all blocks created in this hour "
+  hourlyCumulativeSize: BigInt!
+
+  " Hourly mean size of a block during this hour "
+  hourlyMeanBlockSize: BigDecimal!
+
+  " All of the blocks created during this day "
+  blocks: [Block!]! @derivedFrom(field: "hour")
+}
+
+##########################
+#### Block-level Data ####
+##########################
+
+type Block @entity {
+  " { block number } "
+  id: ID!
+
+  " Hash of this block "
+  hash: String!
+
+  " Timestamp of the block "
+  timestamp: BigInt!
+
+  " Author of this block (the entity that successfully found this block) "
+  author: Author!
+
+  " Data size of the block in bytes "
+  size: BigInt
+
+  " Minimum gasUsed multiplier for transaction to be included (note, post London upgrade) "
+  baseFeePerGas: BigInt
+
+  " Amount of effort required to create this block "
+  difficulty: BigInt!
+
+  " Total difficulty on chain until this block "
+  cumulativeDifficulty: BigInt!
+
+  " Total gas limit provided from all transactions this block "
+  gasLimit: BigInt!
+
+  " Total gas used in this block "
+  gasUsed: BigInt!
+
+  " Percentage of gas filled in this block (0-100)"
+  gasFilledPercentage: BigDecimal!
```

Comment:
shouldn't this be called `blockUtilization` or `blockUsage`?
https://medium.com/@ASvanevik/why-all-these-empty-ethereum-blocks-666acbbf002